pkg/storageincentives/nodestatus.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storageincentives
+
+import (
+	"context"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/log"
+	"github.com/ethersphere/bee/pkg/settlement/swap/erc20"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/storageincentives/redistribution"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"math/big"
+	"sync"
+)
+
+const redistributionStatusKey = "redistribution_state"
+const loggerNameNode = "nodestatus"
+
+type NodeState struct {
+	stateStore     storage.StateStorer
+	erc20Service   erc20.Service
+	logger         log.Logger
+	overlay        swarm.Address
+	mtx            sync.Mutex
+	nodeStatus     NodeStatus
+	currentBalance *big.Int
+	contract       redistribution.Contract
+}
+
+// NodeStatus provide internal status of the nodes in the redistribution game
+type NodeStatus struct {
```

Comment:
the name should ideally be `incentivesStatus` or just `status`, nodeStatus is too general imo

Reply:
Done